tasks/go_test.py

```diff
@@ -366,29 +369,46 @@ def test_flavor(
     junit_tar: str,
     save_result_json: str,
     test_profiler: TestProfiler,
+    coverage_script_template: str,
 ):
     """
     Runs unit tests for given flavor, build tags, and modules.
     """
     args["go_build_tags"] = " ".join(build_tags)
 
-    junit_file_flag = ""
     junit_file = f"junit-out-{flavor.name}.xml"
-    if junit_tar:
-        junit_file_flag = "--junitfile " + junit_file
+    junit_file_flag = "--junitfile " + junit_file if junit_tar else ""
     args["junit_file_flag"] = junit_file_flag
 
     def command(test_results, module, module_result):
         with ctx.cd(module.full_path()):
+            packages = ' '.join(f"{t}/..." if not t.endswith("/...") else t for t in module.targets)
+
+            # Workaround of https://github.com/gotestyourself/gotestsum/issues/274.
+            # Unit tests reruns rewrite the whole coverage file, making it inaccurate.
+            # We use the --raw-command flag to tell each `go test` iteration to write coverage in a different file.
+            cov_test_path = os.path.join(module.full_path(), GO_COV_TEST_PATH)
+            coverage_script = coverage_script_template.format(packages=packages, **args)
+            with open(cov_test_path, 'w', encoding='utf-8') as f:
+                f.write(coverage_script)
+            os.chmod(cov_test_path, 0o755)
+
             res = ctx.run(
-                cmd.format(
-                    packages=' '.join(f"{t}/..." if not t.endswith("/...") else t for t in module.targets), **args
-                ),
+                command=cmd.format(packages=packages, cov_test_path=cov_test_path, **args),
                 env=env,
                 out_stream=test_profiler,
                 warn=True,
             )
 
+        # Removing the coverage script.
+        os.remove(cov_test_path)
+
+        # Merging the unit tests reruns coverage files, keeping only the merged file.
+        ctx.run(f"gocovmerge {TMP_PROFILE_COV_PREFIX}.* > {PROFILE_COV}")
+        for f in os.listdir('.'):
+            if f.startswith(TMP_PROFILE_COV_PREFIX + "."):
```

Comment:
nit, for consistence
```suggestion
            if f.startswith(f"{TMP_PROFILE_COV_PREFIX}."):
```